R/init.R

```diff
@@ -1,15 +1,82 @@
 #' Initialize the R environment with envsetup
 #'
 #' @param project Character. The path to the project directory. Defaults to the current working directory.
-#' @param config_path Character. The path of the config file. Defaults to "envsetup.yml".
+#' @param config_path Character. The path of the config file. Defaults to NULL.
+#' @param create_paths Logical indicating if missing paths should be created. Defaults to NULL.
 #' @export
 #' @return Logical. TRUE if successful.
 #'
 #' @examples
 #' \dontrun{
 #' init()
 #' }
-init <- function(project = getwd(), config_path = "envsetup.yml") {
+init <- function(project = getwd(), config_path = NULL, create_paths = NULL) {
+
+  create_config <- FALSE
+  config_found <- FALSE
+
+  if (is.null(config_path)) {
+    create_config <- usethis::ui_yeah("No path to an exisiting configuration file was provided.
+                                      Would you like us to create a default configuration in your project directory?",
+      n_no = 1
+    )
+  } else {
+    if (file.exists(config_path) && !dir.exists(config_path)) {
+      config_found <- TRUE
+
+      usethis::ui_done("Configuration file found!")
+
+      # verify directories exist
+      config <- config::get(file = config_path)
+
+      if (!exists("paths", where = config)) {
+        usethis::ui_oops("No paths are specified as part of your configuration.  Update your config file to add paths.")
+        return(invisible())
+      }
+
+      paths <- unlist(config$paths, use.names = FALSE)
+
+      missing_directories <- !sapply(paths, dir.exists)
```

Comment:
So if you look at the docs for `sapply()` they indicate that it shouldn't be used in package development. Even though this is a controlled instance, I'd recommend that we replace this with vapply:
```suggestion
      missing_directories <- !vapply(paths, dir.exists, TRUE)
```